### PR TITLE
[salome-medcoupling] Add new port

### PR DESCRIPTION
**Pull request by Neumann-A**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/salome-configuration/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "salome-configuration",
+  "version": "9.10.0",
+  "description": "Configuration files and other utilities for SALOME platform",
+  "homepage": "https://www.salome-platform.org",
+  "license": "LGPL-2.1-or-later"
```
Comment (jimwang118) [inline ports/salome-configuration/vcpkg.json:6]:
```suggestion
  "license": "LGPL-2.1-or-later",
  "supports": "windows | linux"
```

File: ports/salome-medcoupling/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,60 @@
+vcpkg_from_git(
+    OUT_SOURCE_PATH SOURCE_PATH
+    URL "https://git.salome-platform.org/gitpub/tools/medcoupling.git"
+    REF "fe2e38d301902c626f644907e00e499552bb2fa5"
+    PATCHES win.patch fix-missing-symbols.patch
```
Comment (jimwang118) [inline ports/salome-medcoupling/portfile.cmake:5]:
```suggestion
    PATCHES 
        win.patch 
        fix-missing-symbols.patch
```

File: ports/salome-medcoupling/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+)
+
+vcpkg_cmake_install()
+
+file(GLOB dll_files "${CURRENT_PACKAGES_DIR}/lib/*.dll")
+file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/bin")
+foreach(dll_file IN LISTS dll_files)
+  string(REPLACE "/lib/" "/bin/" new_loc "${dll_file}")
+  file(RENAME "${dll_file}" "${new_loc}")
+endforeach()
```
Comment (jimwang118) [inline ports/salome-medcoupling/portfile.cmake:32]: Is this modification method more appropriate to put in the install in CMakeLists.txt?

File: ports/salome-configuration/vcpkg.json (hunk 1 above)
Comment (jimwang118) [reply ports/salome-configuration/vcpkg.json:6]: The upstream indicates that this port supports [Windows and Linux](https://www.salome-platform.org/?p=2446#:~:text=Linux%20Debian%209,Windows%2010).

**Review by jimwang118 — CHANGES_REQUESTED** [review]

**Comment by jimwang118** [comment]
Note: I will be converting your PR to draft status. The above suggested changes are only recommendations. If you are willing to adopt them, please click "ready for review" after making the modifications. If you do not wish to make any changes, please click "ready for review" directly. That way, I can be aware that you've responded since you can't modify the tags.

File: ports/salome-medcoupling/portfile.cmake (hunk 3 above)
Comment (Neumann-A) [reply ports/salome-medcoupling/portfile.cmake:34]: The install scripts do use:
155 ADD_LIBRARY(medpartitionercpp ${medpartitionercpp_SOURCES})
[ 156](https://git.salome-platform.org/gitweb/?p=tools/medcoupling.git;a=blob;f=src/MEDPartitioner/CMakeLists.txt;h=e088a0ed480f87bfc01ed5f00216d06ed5df6ded;hb=fe2e38d301902c626f644907e00e499552bb2fa5#l156) TARGET_LINK_LIBRARIES(medpartitionercpp ${medpartitionercpp_LDFLAGS} ${PLATFORM_LIBS} ${PTHREAD_LIBS})
[ 157](https://git.salome-platform.org/gitweb/?p=tools/medcoupling.git;a=blob;f=src/MEDPartitioner/CMakeLists.txt;h=e088a0ed480f87bfc01ed5f00216d06ed5df6ded;hb=fe2e38d301902c626f644907e00e499552bb2fa5#l157) INSTALL(TARGETS medpartitionercpp DESTINATION ${MEDCOUPLING_INSTALL_LIBS})
[ 158](https://git.salome-platform.org/gitweb/?p=tools/medcoupling.git;a=blob;f=src/MEDPartitioner/CMakeLists.txt;h=e088a0ed480f87bfc01ed5f00216d06ed5df6ded;hb=fe2e38d301902c626f644907e00e499552bb2fa5#l158) 
[ 159](https://git.salome-platform.org/gitweb/?p=tools/medcoupling.git;a=blob;f=src/MEDPartitioner/CMakeLists.txt;h=e088a0ed480f87bfc01ed5f00216d06ed5df6ded;hb=fe2e38d301902c626f644907e00e499552bb2fa5#l159) ADD_EXECUTABLE(medpartitioner medpartitioner.cxx)
[ 160](https://git.salome-platform.org/gitweb/?p=tools/medcoupling.git;a=blob;f=src/MEDPartitioner/CMakeLists.txt;h=e088a0ed480f87bfc01ed5f00216d06ed5df6ded;hb=fe2e38d301902c626f644907e00e499552bb2fa5#l160) TARGET_LINK_LIBRARIES(medpartitioner medpartitionercpp ${medpartitionercpp_LDFLAGS})
[ 161](https://git.salome-platform.org/gitweb/?p=tools/medcoupling.git;a=blob;f=src/MEDPartitioner/CMakeLists.txt;h=e088a0ed480f87bfc01ed5f00216d06ed5df6ded;hb=fe2e38d301902c626f644907e00e499552bb2fa5#l161) INSTALL(TARGETS medpartitioner DESTINATION ${MEDCOUPLING_INSTALL_BINS})

As you can see the install command is missing the `COMPONENTS` keyword which would make this control possible. Furthermore this is just 1 subdir of the 10 subdirs this subproject has. On top of that every salome module has this problem. As such, moving the files is a lot less code than actual patching the tens of CMakeLists.txt

**Review by jimwang118 — APPROVED** [review]

**Comment by jimwang118** [comment]
Usage test pass with following triplets:

```
x86-windows
x64-windows
```

**Comment by JonLiu1993** [comment]
@Neumann-A, There is an error in CI because the salome-medcoupling download link https://git.salome-platform.org/gitpub/tools/configuration.git cannot be opened. I would like to ask if there are other download links for this port that can be replaced?
````
fatal: unable to access 'https://git.salome-platform.org/gitpub/tools/configuration.git/': Could not resolve host: git.salome-platform.org
````

**Comment by Neumann-A** [comment]
@JonLiu1993 I think there web server just went down. Wait a week an it probably will be back.

**Pull request merged**